astropy/coordinates/errors.py

```diff
@@ -31,3 +67,25 @@ def __init__(self, site, attribute, close_names=None):
         self.attribute = attribute
         self.close_names = close_names
         return super().__init__(message)
+
+
+class NonRotationTransformationWarning(AstropyUserWarning):
+    """
+    Emitted for transformations that are not simple rotations. Such
+    transformations can change the angular separation between coordinates
+    depending on its direction.
+    """
+
+    def __init__(
+        self, frame_to: BaseCoordinateFrame, frame_from: BaseCoordinateFrame
+    ) -> None:
+        self.frame_to = frame_to
```

Comment:
I'd tend to swap these two, but no big deal to keep it as is (I can see the nicety of `self, other_frame` in the actual initialization...)

Reply:
I prefer to keep things convenient for the caller, so I'm leaving this as is.